File: rust/src/models/product_family_list_response.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::{pagination_response::PaginationResponse, product_family::ProductFamily};

#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct ProductFamilyListResponse {
    pub data: Vec<ProductFamily>,

    pub pagination_meta: PaginationResponse,
}

impl ProductFamilyListResponse {
    pub fn new(data: Vec<ProductFamily>, pagination_meta: PaginationResponse) -> Self {
        Self {
            data,
            pagination_meta,
        }
    }
}

File: typescript/src/models/productFamilyListResponse.ts
// this file is @generated
import {
  type PaginationResponse,
  PaginationResponseSerializer,
} from "./paginationResponse";
import { type ProductFamily, ProductFamilySerializer } from "./productFamily";

export interface ProductFamilyListResponse {
  data: ProductFamily[];

  paginationMeta: PaginationResponse;
}

export const ProductFamilyListResponseSerializer = {
  _fromJsonObject(object: any): ProductFamilyListResponse {
    return {
      data: object["data"].map((item: any) =>
        ProductFamilySerializer._fromJsonObject(item)
      ),
      paginationMeta: PaginationResponseSerializer._fromJsonObject(
        object["pagination_meta"]
      ),
    };
  },

  _toJsonObject(self: ProductFamilyListResponse): any {
    return {
      data: self.data.map((item: any) => ProductFamilySerializer._toJsonObject(item)),
      pagination_meta: PaginationResponseSerializer._toJsonObject(self.paginationMeta),
    };
  },
};
